include/alpaka/rand/Philox/PhiloxSingle.hpp

```diff
@@ -84,12 +84,23 @@ namespace alpaka
                  */
                 ALPAKA_FN_HOST_ACC auto nextNumber()
                 {
+                    // Element zero will always contain the next valid random number.
+                    auto result = state.result[0];
                     state.position++;
```

Comment:
@sliwowitz During the first access this line increments the position from `0` to `1` and `return state.result[state.position];` will therefore always skip the value in the array on pos index 0.

Reply:
All elements will be used because the access to the rng number is after the advance function where the position is set back to zero.